web/html/src/components/input/InputBase.js

```diff
@@ -75,20 +78,42 @@ class InputBase extends React.Component<Props, State> {
     });
   }
 
-  setValue(name: string, value: string) {
-    if (this.props.onFormChange) {
-      this.props.onFormChange({
-        name,
-        value,
-      });
+  validate(model: Object): boolean {
+    const { name } = this.props;
+    const results = [];
+    let isValid = true;
+
+    if (!this.props.disabled && (model[name] || this.props.required)) {
+      if (this.props.required && !model[name]) {
+        isValid = false;
+      } else if (this.props.validators) {
+        const validators = Array.isArray(this.props.validators) ? this.props.validators : [this.props.validators];
+        validators.forEach((v) => {
+          results.push(Promise.resolve(v(`${model[name] || ''}`)));
+        });
+      }
     }
 
+    this.setState({
+      isValid,
+    });
+    return isValid;
+  }
+
+  setValue(name: string, value: string) {
+    const { model } = this.context;
+    model[name] = value;
```

Comment:
I think these lines shouldn't be here since `setModelValue` does the job properly below